File: leetcode/2966. Divide Array Into Arrays With Max Difference.cpp
class Solution {
public:
    vector<vector<int>> divideArray(vector<int>& nums, int k) {
        int n = nums.size();
        sort(nums.begin(), nums.end());
        vector<vector<int>> ans;

        for (int i = 0; i < n; i += 3) {
            if ((nums[i + 1] - nums[i]) > k ||
                (nums[i + 2] - nums[i + 1]) > k ||
                (nums[i + 2] - nums[i]) > k) {
                return {};
            }

            ans.push_back(vector<int>{nums[i], nums[i + 1], nums[i + 2]});
        }

        return ans;
    }
};

File: leetcode/2966. Divide Array Into Arrays With Max Difference.java
class Solution {
    public int[][] divideArray(int[] nums, int k) {
        Arrays.sort(nums);
        int ansInx = 0, n = nums.length;
        int[][] ans = new int[n / 3][];

        for (int i = 0; i < n - 2; i += 3) {
            if ((nums[i + 1] - nums[i]) > k ||
                    (nums[i + 2] - nums[i]) > k ||
                    (nums[i + 2] - nums[i + 1]) > k) {
                return new int[][] {};
            }

            ans[ansInx++] = new int[] { nums[i], nums[i + 1], nums[i + 2] };
        }
        return ans;
    }
}